tests/framework/filters/AccessRuleTest.php

```diff
@@ -88,7 +145,107 @@ public function testMatchAction()
 
     // TODO test match controller
 
-    // TODO test match roles
+    public function testMatchRole()
```

Comment:
thanks for adding tests. Can you transform the tests into PhpUnits dataprovider to benefit from its advantages?